management/src/main/java/team/proximity/management/controllers/ProviderServiceController.java

```diff
@@ -28,104 +33,109 @@ public class ProviderServiceController {
     public ProviderServiceController(ProviderServiceService providerServiceService) {
         this.providerServiceService = providerServiceService;
     }
+    @PostMapping(consumes = "multipart/form-data")
+    @Operation(summary = "Create or update a provider service", description = "Creates or updates a provider service")
+    @ApiResponses(value = {
+            @io.swagger.v3.oas.annotations.responses.ApiResponse(responseCode = "204", description = "Successfully deleted service experience"),
+            @ApiResponse(responseCode = "201", description = "Successfully created or updated provider service",
+                    content = @Content(mediaType = "application/json", schema = @Schema(implementation = ProviderService.class))),
+            @ApiResponse(responseCode = "400", description = "Invalid input"),
+            @ApiResponse(responseCode = "401", description = "You are not authorized to view the resource"),
+            @ApiResponse(responseCode = "403", description = "Accessing the resource you were trying to reach is forbidden"),
+            @ApiResponse(responseCode = "404", description = "The resource you were trying to reach is not found")
+    })
+    public ResponseEntity<ApiSuccessResponse<ProviderService>> createOrUpdateProviderService(@Validated @ModelAttribute ProviderServiceRequest providerServiceRequest) throws JsonProcessingException {
+        log.info("Processing providerService request: {}", providerServiceRequest);
 
+        ProviderService providerService = providerServiceService.createOrUpdateProviderService(providerServiceRequest);
 
-
-    @PutMapping("/{id}")
-    @Operation(summary = "Update a providerService", description = "Update a providerService by id")
-
-    public ResponseEntity<ApiResponse<ProviderService>> updateProviderService(@PathVariable UUID id, @RequestBody ProviderServiceRequest providerService) {
```

Comment:
Move all logics to the service. clean up the controller. 
Single Responsibility Principle